ros_ws/src/get_user_request.py

```diff
@@ -0,0 +1,141 @@
+            break
+        context.append(response)
+
+def Judge(request):
+    #llm = OpenAI(model_name=model_to_use, temperature=0)
+    llm = ChatOpenAI(model_name=model_to_use,temperature=0)
+
+    context="""An agent is attempting to get a specific item requested by a human user.  
+        It should be an object found in a house or apartment.
+        The human user is sometimes vague and the agent is trying to clarify specifically what the 
+        human user wants.  You are an AI helping the agent to judge if the item 
+        requested is specific enough to search for.  
+        You are also an expert in grammatical structures and will be given a sentence from which 
+        you should extract what exactly is being requested. It will likely be the direct object in a request 
+        or question or the subject of a statement.  
+        If the response has the word "anything" or "something" in it, it is not specific enough.
+        If the object is specific enough to search for, respond  with "yes", otherwise respond with "no".  Do not be verbose. 
+        """
+    
+    system_message_prompt=SystemMessagePromptTemplate.from_template(context)
+    human_template=f"Am I specifically requesting an object, if so what? : {request}"
+    human_message_prompt=HumanMessagePromptTemplate.from_template(human_template)
+
+    chat_prompt=ChatPromptTemplate.from_messages([system_message_prompt,human_message_prompt])
+    chain=LLMChain(llm=llm, prompt=chat_prompt)
+    response=chain.run(question=request)
+    response=cleanup(response)
+    return response
+
+def extract_request(verbose_request):
+    #llm = OpenAI(model_name=model_to_use, temperature=0)
+    llm = ChatOpenAI(model_name=model_to_use,temperature=0)
+
+    context=""" You are an expert in grammatical structures and will be given a sentence from which 
+        you should extract what exactly is being requested. It will likely be the direct object in a 
+        request or question or the subject of a statement.
+        It should also be an object commonly found in a house or apartment.   
+        Reply with just the object and its modifiers.  
+        For example reply with "can of soda" rahther than just "soda"
+        Do not be verbose."""
+    
+    system_message_prompt=SystemMessagePromptTemplate.from_template(context)
+    human_template=f"Isolate what I am requesting: {verbose_request}"
+    human_message_prompt=HumanMessagePromptTemplate.from_template(human_template)
+    
+
+    chat_prompt=ChatPromptTemplate.from_messages([system_message_prompt,human_message_prompt])
+    chain=LLMChain(llm=llm, prompt=chat_prompt)
+    response=chain.run(question=verbose_request)
+    response=cleanup(response)
+
+    return response
+
+def cleanup(response):
+    response=response.replace("\n","")
+    response=response.replace("Answer: ","")
+    response=response.replace(".","")
+    return response
+
+if __name__=="__main__":
+    #request="Please get me a bottle of water"
```

Comment:
Remove when we get closer to producing this as a finalized service; no extraneous code.